sdk/core/azure-core/inc/azure/core/url.hpp

```diff
@@ -239,5 +239,55 @@ namespace Azure { namespace Core {
      * @return Absolute URL with URL-encoded query parameters.
      */
     std::string GetAbsoluteUrl() const;
+
+    /**
+     * @brief Compares with another instance of Url for equality.
+     * @param other another instance of Url.
+     * @return `true` if values of two Url instances are equal, `false` otherwise.
+     *
+     */
+    bool operator==(Url const& other) const
+    {
+      if (this == &other)
+      {
+        return true;
+      }
+
+      if (m_port != other.m_port
```

Comment:
For a scheme based normalization, this check would mean an 80 port and no port won't be considered equal.

I am calling this out as one example of where we need to figure out our strategy here, assuming we even want to go down this path of adding this capability to the Url class.

We'd want test and comments around this.


> because the "http" scheme makes use of an authority component, has a
   default port of "80", and defines an empty path to be equivalent to
   "/", the following four URIs are equivalent:

      http://example.com
      http://example.com/
      http://example.com:/
      http://example.com:80/

Reply:
I added a comment clarifying that equality means string representation equality, as well as included the link to an RFC section 6.2.1

I think it might be useful to have normalizations, but I think we should bring them to `ToString()` method as well, and then update the equality comparison implementation if needed.

Reply:
BTW, the very first 3 examples that you mentioned are indistinguishable with our implementation, after they are parsed, all going to have exactly the same `ToString()`, and our SDK would eventually convert these URLs to strings - either as a URL to a transport adapter, or a serialized payload to Azure.